File: core/src/com/desitum/library/view/ViewGroup.java
package com.desitum.library.view;

import com.desitum.library.game.World;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by kodyvanry on 5/1/17.
 */

public class ViewGroup extends View {

    private List<View> mChildren;

    public ViewGroup(World world) {
        this(world, null);
    }

    public ViewGroup(World world, LayoutConstraints layoutConstraints) {
        super(world, layoutConstraints);
        mChildren = new ArrayList<View>();
    }

    public List<View> getChildren() {
        return mChildren;
    }

    public void setChildren(List<View> children) {
        this.mChildren = children;
    }

    public void addView(View v) {
        mChildren.add(v);
    }

    public void removeView(View v) {
        mChildren.remove(v);
        getWorld().removeView(v);
    }

    @Override
    public void invalidate() {
        super.invalidate();
        for (View v : mChildren) {
            v.invalidate();
        }
    }
}

File: core/src/com/desitum/library/view/ViewGroup.kt
package com.desitum.library.view

import com.desitum.library.game.World

import java.util.ArrayList

/**
 * Created by kodyvanry on 5/1/17.
 */

open class ViewGroup @JvmOverloads constructor(world: World, layoutConstraints: LayoutConstraints? = null) : View(world, layoutConstraints) {

    var children: MutableList<View> = ArrayList<View>()

    open fun addView(v: View) {
        children.add(v)
        v.z = z + 1
        v.parent = this
        world?.addView(v)
    }

    open fun removeView(v: View) {
        children.remove(v)
        world?.removeView(v)
    }

    override fun invalidate() {
        super.invalidate()
        children.forEach { it.invalidate() }
    }

    fun hideViews(vararg ids: String) {
        ids.forEach {
        }
    }
}
